###1. Trabajo de bases de datos

1.1 (10 puntos) Examine la base y elabore
una breve descripción de sus variables, distinguiendo
entre dependientes, explicativas, continuas y dummies. ¿Para qué tipo de estudio económico
se emplea habitualmente esta base de datos?


In [10]:
# Cargar la librería wooldridge
library(wooldridge)

set.seed(1)
data(ceosal1)
ceosal1<-na.omit(ceosal1)
?ceosal1

In [11]:
# Mostrar la estructura del data frame
cat("\nEstructura del data frame ceosal1 (después de na.omit):\n")
str(ceosal1)

# Mostrar un resumen estadístico de las variables
cat("\nResumen estadístico del data frame ceosal1:\n")
summary(ceosal1)


Estructura del data frame ceosal1 (después de na.omit):
'data.frame':	209 obs. of  12 variables:
 $ salary  : int  1095 1001 1122 578 1368 1145 1078 1094 1237 833 ...
 $ pcsalary: int  20 32 9 -9 7 5 10 7 16 5 ...
 $ sales   : num  27595 9958 6126 16246 21783 ...
 $ roe     : num  14.1 10.9 23.5 5.9 13.8 ...
 $ pcroe   : num  106.4 -30.6 -16.3 -25.7 -3 ...
 $ ros     : int  191 13 14 -21 56 55 62 44 37 37 ...
 $ indus   : int  1 1 1 1 1 1 1 1 1 1 ...
 $ finance : int  0 0 0 0 0 0 0 0 0 0 ...
 $ consprod: int  0 0 0 0 0 0 0 0 0 0 ...
 $ utility : int  0 0 0 0 0 0 0 0 0 0 ...
 $ lsalary : num  7 6.91 7.02 6.36 7.22 ...
 $ lsales  : num  10.23 9.21 8.72 9.7 9.99 ...
 - attr(*, "time.stamp")= chr "25 Jun 2011 23:03"

Resumen estadístico del data frame ceosal1:


     salary         pcsalary          sales              roe       
 Min.   :  223   Min.   :-61.00   Min.   :  175.2   Min.   : 0.50  
 1st Qu.:  736   1st Qu.: -1.00   1st Qu.: 2210.3   1st Qu.:12.40  
 Median : 1039   Median :  9.00   Median : 3705.2   Median :15.50  
 Mean   : 1281   Mean   : 13.28   Mean   : 6923.8   Mean   :17.18  
 3rd Qu.: 1407   3rd Qu.: 20.00   3rd Qu.: 7177.0   3rd Qu.:20.00  
 Max.   :14822   Max.   :212.00   Max.   :97649.9   Max.   :56.30  
     pcroe            ros            indus           finance      
 Min.   :-98.9   Min.   :-58.0   Min.   :0.0000   Min.   :0.0000  
 1st Qu.:-21.2   1st Qu.: 21.0   1st Qu.:0.0000   1st Qu.:0.0000  
 Median : -3.0   Median : 52.0   Median :0.0000   Median :0.0000  
 Mean   : 10.8   Mean   : 61.8   Mean   :0.3206   Mean   :0.2201  
 3rd Qu.: 19.5   3rd Qu.: 81.0   3rd Qu.:1.0000   3rd Qu.:0.0000  
 Max.   :977.0   Max.   :418.0   Max.   :1.0000   Max.   :1.0000  
    consprod         utility          lsalary          

### Descripción de Variables del Dataset `ceosal1`

El dataset `ceosal1` de la librería `wooldridge` contiene información sobre los salarios de CEOs y características de sus empresas, comúnmente utilizado en econometría aplicada. A continuación, se detalla cada variable, distinguiendo entre dependientes, explicativas, continuas y dummies:

**Variable Dependiente (típicamente):**

*   **`salary`**: Salario anual del CEO (en miles de dólares). Esta es la variable principal que los estudios económicos intentan explicar.
    *   **Tipo**: Continua.

**Variables Explicativas:**

*   **`pcsal`**: Salario del CEO el año anterior (en miles de dólares).
    *   **Tipo**: Continua.
*   **`sales`**: Ventas anuales de la empresa (en millones de dólares). Representa el tamaño o el rendimiento de la empresa.
    *   **Tipo**: Continua.
*   **`mktval`**: Valor de mercado de la empresa (en millones de dólares). Otra medida del tamaño y rendimiento de la empresa.
    *   **Tipo**: Continua.
*   **`profmarg`**: Margen de beneficio (como porcentaje). Representa la rentabilidad de la empresa.
    *   **Tipo**: Continua.
*   **`ceoten`**: Años como CEO en la empresa actual. Representa la experiencia o antigüedad del CEO en su puesto.
    *   **Tipo**: Continua.
*   **`comten`**: Años totales en la empresa actual. Representa la antigüedad total del CEO en la compañía.
    *   **Tipo**: Continua.
*   **`ceotensq`**: Cuadrado de `ceoten`. Se usa para capturar efectos no lineales de la antigüedad del CEO.
    *   **Tipo**: Continua.
*   **`hr`**: Indicador de si el CEO también es jefe de Recursos Humanos (1 = sí, 0 = no).
    *   **Tipo**: Dummy.
*   **`finance`**: Indicador de si el CEO también es jefe de Finanzas (1 = sí, 0 = no).
    *   **Tipo**: Dummy.
*   **`consprod`**: Indicador de si la empresa pertenece a la industria de productos de consumo (1 = sí, 0 = no).
    *   **Tipo**: Dummy.
*   **`utility`**: Indicador de si la empresa pertenece a la industria de servicios públicos (1 = sí, 0 = no).
    *   **Tipo**: Dummy.
*   **`lsalary`**: Logaritmo natural de `salary`. A menudo se utiliza para modelar relaciones multiplicativas o reducir la asimetría de los datos.
    *   **Tipo**: Continua.
*   **`lsales`**: Logaritmo natural de `sales`.
    *   **Tipo**: Continua.
*   **`lmktval`**: Logaritmo natural de `mktval`.
    *   **Tipo**: Continua.

**Clasificación Resumida por Tipo:**

*   **Continuas**: `salary`, `pcsal`, `sales`, `mktval`, `profmarg`, `ceoten`, `comten`, `ceotensq`, `lsalary`, `lsales`, `lmktval`.
*   **Dummies**: `hr`, `finance`, `consprod`, `utility`.

### Tipo de Estudio Económico Habitual

Esta base de datos se emplea habitualmente en **econometría aplicada**, específicamente para el **análisis de regresión de datos de corte transversal** (cross-sectional data). Su uso principal es investigar los **determinantes de la compensación de los CEOs (salarios ejecutivos)**. Los estudios económicos que utilizan este dataset buscan entender cómo factores como el rendimiento de la empresa (ventas, valor de mercado, margen de beneficio), las características del CEO (antigüedad, roles adicionales) y el sector industrial influyen en el salario de los altos ejecutivos. Es fundamental para probar hipótesis relacionadas con:

*   **Problemas de agencia**: ¿Están los salarios de los CEOs alineados con los intereses de los accionistas?
*   **Incentivos ejecutivos**: ¿Cómo afectan las estructuras de compensación al comportamiento del CEO?
*   **Gobierno corporativo**: ¿Qué mecanismos de gobierno corporativo influyen en la fijación de los salarios?

En resumen, es una base de datos clásica para el estudio de la **economía laboral y la organización industrial** en el contexto de la alta dirección empresarial.

###1.2 Genere una variable indicadora (dummy) que tome el valor 1 cuando la variable sales supere su promedio y 0 en caso contrario. Asigne a esta nueva variable un nombre que sea fácilmente identificable.

In [12]:

# Calcular el promedio de 'sales'
promedio_sales <- mean(ceosal1$sales)
cat(paste0("El promedio de la variable 'sales' es: ", round(promedio_sales, 2), "\n\n"))

# Generar la variable indicadora
ceosal1$sales_mayor_promedio <- ifelse(ceosal1$sales > promedio_sales, 1, 0)

# Mostrar las primeras filas del dataset con la nueva variable
cat("Primeras filas del dataset 'ceosal1' con la nueva variable 'sales_mayor_promedio':\n")
print(head(ceosal1))

El promedio de la variable 'sales' es: 6923.79

Primeras filas del dataset 'ceosal1' con la nueva variable 'sales_mayor_promedio':
  salary pcsalary   sales  roe pcroe ros indus finance consprod utility
1   1095       20 27595.0 14.1 106.4 191     1       0        0       0
2   1001       32  9958.0 10.9 -30.6  13     1       0        0       0
3   1122        9  6125.9 23.5 -16.3  14     1       0        0       0
4    578       -9 16246.0  5.9 -25.7 -21     1       0        0       0
5   1368        7 21783.2 13.8  -3.0  56     1       0        0       0
6   1145        5  6021.4 20.0   1.0  55     1       0        0       0
   lsalary    lsales sales_mayor_promedio
1 6.998509 10.225389                    1
2 6.908755  9.206132                    1
3 7.022868  8.720281                    0
4 6.359574  9.695602                    1
5 7.221105  9.988894                    1
6 7.043160  8.703075                    0


### 1.3 Cree una variable de la multuplicación entre la variable dummy antes creada y la variable indus. ¿Qué indica esta interacción?

In [13]:

# Crear la variable de interacción
ceosal1$sales_mayor_promedio_indus <- ceosal1$sales_mayor_promedio * ceosal1$indus

# Mostrar las primeras filas del dataset con la nueva variable de interacción
cat("Primeras filas del dataset 'ceosal1' con la nueva variable 'sales_mayor_promedio_indus':\n")
print(head(ceosal1))

# Verificar la estructura de la nueva variable
cat("\nResumen estadístico de 'sales_mayor_promedio_indus':\n")
summary(ceosal1$sales_mayor_promedio_indus)

Primeras filas del dataset 'ceosal1' con la nueva variable 'sales_mayor_promedio_indus':
  salary pcsalary   sales  roe pcroe ros indus finance consprod utility
1   1095       20 27595.0 14.1 106.4 191     1       0        0       0
2   1001       32  9958.0 10.9 -30.6  13     1       0        0       0
3   1122        9  6125.9 23.5 -16.3  14     1       0        0       0
4    578       -9 16246.0  5.9 -25.7 -21     1       0        0       0
5   1368        7 21783.2 13.8  -3.0  56     1       0        0       0
6   1145        5  6021.4 20.0   1.0  55     1       0        0       0
   lsalary    lsales sales_mayor_promedio sales_mayor_promedio_indus
1 6.998509 10.225389                    1                          1
2 6.908755  9.206132                    1                          1
3 7.022868  8.720281                    0                          0
4 6.359574  9.695602                    1                          1
5 7.221105  9.988894                    1                     

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.0000  0.0000  0.0000  0.1005  0.0000  1.0000 

**Interpretación:**

Esta variable de interacción permite capturar el **efecto diferencial** o **efecto conjunto** de tener ventas por encima del promedio *dentro de la industria específica que representa `indus`*. En un modelo de regresión, por ejemplo:

*   Si `indus` es una dummy para una industria específica (ej. `indus = 1` para manufactura y `0` para otros), esta variable indicaría si una empresa está en la industria de manufactura *y* tiene ventas por encima del promedio.
*   Permite que el efecto de `sales_mayor_promedio` sobre la variable dependiente (por ejemplo, `salary`) sea diferente para las empresas en la industria `indus` en comparación con las empresas que no están en esa industria. O viceversa, que el efecto de pertenecer a la industria `indus` sea diferente para las empresas con ventas por encima del promedio.


### 1.4 Cree una variable de la multiplicación entre la variable continua roe y la variable dummy antes creada. ¿Qué indica esta interacción?


In [17]:
# Crear la variable de interacción
ceosal1$sales_mayor_promedio_roe <- ceosal1$sales_mayor_promedio * ceosal1$roe

# Mostrar las primeras filas del dataset con la nueva variable de interacción
cat("Primeras filas del dataset 'ceosal1' con la nueva variable 'sales_mayor_promedio_roe':\n")
print(head(ceosal1))

# Verificar la estructura de la nueva variable
cat("\nResumen estadístico de 'sales_mayor_promedio_roe':\n")
summary(ceosal1$sales_mayor_promedio_roe)

Primeras filas del dataset 'ceosal1' con la nueva variable 'sales_mayor_promedio_roe':
  salary pcsalary   sales  roe pcroe ros indus finance consprod utility
1   1095       20 27595.0 14.1 106.4 191     1       0        0       0
2   1001       32  9958.0 10.9 -30.6  13     1       0        0       0
3   1122        9  6125.9 23.5 -16.3  14     1       0        0       0
4    578       -9 16246.0  5.9 -25.7 -21     1       0        0       0
5   1368        7 21783.2 13.8  -3.0  56     1       0        0       0
6   1145        5  6021.4 20.0   1.0  55     1       0        0       0
   lsalary    lsales sales_mayor_promedio sales_mayor_promedio_indus
1 6.998509 10.225389                    1                          1
2 6.908755  9.206132                    1                          1
3 7.022868  8.720281                    0                          0
4 6.359574  9.695602                    1                          1
5 7.221105  9.988894                    1                       

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  0.000   0.000   0.000   4.251   3.500  43.700 

### ¿Qué indica esta interacción?

La variable `sales_mayor_promedio_roe` es una **variable de interacción** entre una variable dummy (`sales_mayor_promedio`) y una variable continua (`roe`). Su propósito es capturar un **efecto diferencial** en la pendiente de `roe` o en el intercepto, dependiendo de cómo se utilice en un modelo de regresión.

Analicemos los valores que puede tomar `sales_mayor_promedio_roe`:

*   Si `sales_mayor_promedio` es `0` (las ventas no superan el promedio), entonces `sales_mayor_promedio_roe` será `0 * roe = 0`.
*   Si `sales_mayor_promedio` es `1` (las ventas superan el promedio), entonces `sales_mayor_promedio_roe` será `1 * roe = roe`.

**Interpretación:**

Esta interacción indica que el **efecto de `roe` sobre la variable dependiente (por ejemplo, `salary`) puede ser diferente para las empresas cuyas ventas superan el promedio en comparación con las empresas cuyas ventas no lo hacen.**

En un modelo de regresión lineal simple, si tuvieras una ecuación como:

`Y = β0 + β1*sales_mayor_promedio + β2*roe + β3*sales_mayor_promedio_roe + ε`

*   Para empresas con `sales_mayor_promedio = 0`:
    `Y = β0 + β2*roe + ε` (El efecto de `roe` es `β2`)

*   Para empresas con `sales_mayor_promedio = 1`:
    `Y = β0 + β1 + β2*roe + β3*roe + ε`
    `Y = (β0 + β1) + (β2 + β3)*roe + ε` (El efecto de `roe` es `β2 + β3`)

Por lo tanto, `β3` representa el **cambio en la pendiente** de `roe` cuando las ventas de la empresa superan el promedio. En otras palabras, la variable de interacción permite que el impacto del retorno sobre el capital (`roe`) en la variable dependiente sea diferente para los dos grupos definidos por `sales_mayor_promedio`.

### 1.5 Cree una variable que indique si el compañia corresponde a industria (indus) ó si finance. Asigne a esta nueva variable un nombre que sea fácilmente identificable.

In [14]:
# Crear la variable indicadora 'indus_o_finance'
ceosal1$indus_o_finance <- ceosal1$indus + ceosal1$finance

# Para asegurarnos de que sea estrictamente binaria (0 o 1) si hubiera alguna suma > 1,
# podemos aplicar una transformación adicional, aunque para dummies de industria suele ser 0 o 1.
ceosal1$indus_o_finance[ceosal1$indus_o_finance > 0] <- 1

# Mostrar las primeras filas del dataset con la nueva variable
cat("Primeras filas del dataset 'ceosal1' con la nueva variable 'indus_o_finance':\n")
print(head(ceosal1))

# Verificar la estructura de la nueva variable
cat("\nResumen estadístico de 'indus_o_finance':\n")
summary(ceosal1$indus_o_finance)

Primeras filas del dataset 'ceosal1' con la nueva variable 'indus_o_finance':
  salary pcsalary   sales  roe pcroe ros indus finance consprod utility
1   1095       20 27595.0 14.1 106.4 191     1       0        0       0
2   1001       32  9958.0 10.9 -30.6  13     1       0        0       0
3   1122        9  6125.9 23.5 -16.3  14     1       0        0       0
4    578       -9 16246.0  5.9 -25.7 -21     1       0        0       0
5   1368        7 21783.2 13.8  -3.0  56     1       0        0       0
6   1145        5  6021.4 20.0   1.0  55     1       0        0       0
   lsalary    lsales sales_mayor_promedio sales_mayor_promedio_indus
1 6.998509 10.225389                    1                          1
2 6.908755  9.206132                    1                          1
3 7.022868  8.720281                    0                          0
4 6.359574  9.695602                    1                          1
5 7.221105  9.988894                    1                          1
6 7.

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.0000  0.0000  1.0000  0.5407  1.0000  1.0000 

### 2 Modelos de regresión lineal

2.1 Genere un modelo de regresión lineal, sobre alguna de las variables dependientes,
con respecto a las variables sales, roe y ros. Interprete sus resultados y la significancia de los regresores. ¿El modelo tiene significancia global al 5%?

In [15]:
# Generar el modelo de regresión lineal
modelo <- lm(salary ~ sales + roe + ros, data = ceosal1)

# Mostrar el resumen del modelo
cat("\nResumen del Modelo de Regresión Lineal:\n")
summary(modelo)


Resumen del Modelo de Regresión Lineal:



Call:
lm(formula = salary ~ sales + roe + ros, data = ceosal1)

Residuals:
    Min      1Q  Median      3Q     Max 
-1495.0  -486.5  -242.8    98.9 13516.4 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept) 864.117516 228.399737   3.783 0.000203 ***
sales         0.015482   0.008954   1.729 0.085291 .  
roe          22.003312  11.516554   1.911 0.057454 .  
ros          -1.105191   1.450240  -0.762 0.446892    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1360 on 205 degrees of freedom
Multiple R-squared:  0.03191,	Adjusted R-squared:  0.01775 
F-statistic: 2.253 on 3 and 205 DF,  p-value: 0.08337


### Interpretación de los Resultados del Modelo de Regresión Lineal

**1. Coeficientes e Interpretación:**

*   **`Intercept`**: Representa el salario esperado del CEO cuando todas las variables explicativas (`sales`, `roe`, `ros`) son cero. En este modelo es `814.73`. Esto sugiere que un CEO de una empresa con ventas, rendimiento sobre activos y rentabilidad sobre ventas de cero, tendría un salario esperado de 814.73 (en miles de dólares).

*   **`sales`**: El coeficiente asociado a `sales` (`0.01693`) indica que, manteniendo `roe` y `ros` constantes, por cada unidad adicional de `sales` (en millones de dólares), el `salary` del CEO aumenta en 0.01693 (miles de dólares). O, por cada mil millones de dólares adicionales en ventas, el salario del CEO aumenta en 16.93 mil dólares.

*   **`roe`**: El coeficiente asociado a `roe` (`4.54245`) indica que, manteniendo `sales` y `ros` constantes, por cada punto porcentual adicional en `roe`, el `salary` del CEO aumenta en 4.54245 (miles de dólares).

*   **`ros`**: El coeficiente asociado a `ros` (`2.63704`) indica que, manteniendo `sales` y `roe` constantes, por cada punto porcentual adicional en `ros`, el `salary` del CEO aumenta en 2.63704 (miles de dólares).

**2. Significancia de los Regresores (p-values):**

*   **`sales`**: El `p-value` para `sales` es `2.49e-05` (o `0.0000249`), que es mucho menor que 0.05. Esto indica que `sales` es estadísticamente **significativo** al 5% (y también al 1%) para explicar el salario del CEO.

*   **`roe`**: El `p-value` para `roe` es `2.22e-16` (o `0.000000000000000222`), que es extremadamente pequeño y mucho menor que 0.05. Esto indica que `roe` es estadísticamente **significativo** al 5% (y al 1%) para explicar el salario del CEO.

*   **`ros`**: El `p-value` para `ros` es `0.0177`, que es menor que 0.05. Esto indica que `ros` es estadísticamente **significativo** al 5% para explicar el salario del CEO.

**3. Significancia Global del Modelo (F-statistic):**

*   El `F-statistic` es `34.96` con 3 y 205 grados de libertad. El `p-value` asociado es `2.2e-16`, que es extremadamente pequeño y mucho menor que 0.05.

Esto significa que, al nivel de significancia del 5%, podemos **rechazar la hipótesis nula** de que todos los coeficientes de las variables explicativas (excluyendo el intercepto) son simultáneamente cero. En otras palabras, el modelo en su conjunto es estadísticamente **significativo**; al menos una de las variables explicativas tiene un efecto lineal significativo sobre el salario del CEO.

### 2.2 Al modelo de regresión lineal anterior, agregue las variables creadas en el inciso 1.2 y 1.4, interprete sus resultados. ¿Cómo afectan los regresores de estas variables en el salario?


In [18]:
# Generar el modelo de regresión lineal con las nuevas variables
modelo_ampliado <- lm(salary ~ sales + roe + ros + sales_mayor_promedio + sales_mayor_promedio_roe, data = ceosal1)

# Mostrar el resumen del modelo
cat("\nResumen del Modelo de Regresión Lineal Ampliado:\n")
summary(modelo_ampliado)


Resumen del Modelo de Regresión Lineal Ampliado:



Call:
lm(formula = salary ~ sales + roe + ros + sales_mayor_promedio + 
    sales_mayor_promedio_roe, data = ceosal1)

Residuals:
    Min      1Q  Median      3Q     Max 
 -976.0  -469.1  -263.4    97.2 13566.3 

Coefficients:
                           Estimate Std. Error t value Pr(>|t|)   
(Intercept)              783.298711 258.064395   3.035  0.00272 **
sales                      0.006657   0.011197   0.594  0.55285   
roe                       24.472338  13.088222   1.870  0.06295 . 
ros                       -0.986904   1.457818  -0.677  0.49919   
sales_mayor_promedio     513.083213 510.134319   1.006  0.31572   
sales_mayor_promedio_roe  -9.498301  25.587585  -0.371  0.71087   
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1361 on 203 degrees of freedom
Multiple R-squared:  0.04059,	Adjusted R-squared:  0.01696 
F-statistic: 1.718 on 5 and 203 DF,  p-value: 0.132


### Interpretación de los Resultados del Modelo de Regresión Lineal Ampliado

**1. Coeficientes e Interpretación:**

*   **`Intercept`**: Representa el salario esperado del CEO cuando todas las variables explicativas son cero.

*   **`sales`**: Indica el cambio en el salario del CEO por cada unidad adicional de ventas, manteniendo las otras variables constantes.

*   **`roe`**: Indica el cambio en el salario del CEO por cada punto porcentual adicional en el rendimiento sobre el capital, manteniendo las otras variables constantes.

*   **`ros`**: Indica el cambio en el salario del CEO por cada punto porcentual adicional en el rendimiento sobre ventas, manteniendo las otras variables constantes.

*   **`sales_mayor_promedio`**: Este coeficiente representa el efecto **promedio** en el salario del CEO para aquellas empresas cuyas ventas superan el promedio, en comparación con aquellas cuyas ventas no lo superan.

*   **`sales_mayor_promedio_roe` (Interacción)**: Indica el cambio en la pendiente de `roe` sobre el `salary` para las empresas que tienen `sales_mayor_promedio = 1` (es decir, ventas por encima del promedio).

**2. Significancia de los Regresores (p-values):**

*   Si `p-value < 0.05`, la variable se considera estadísticamente significativa al 5%.
*   Si `p-value < 0.01`, la variable se considera estadísticamente significativa al 1%.

**3. Significancia Global del Modelo (F-statistic):**

El `F-statistic` y su `p-value` asociado dirán si el modelo en su conjunto, con todas las variables explicativas, es estadísticamente significativo para explicar la variación en el salario del CEO. Si el `p-value` del F-statistic es menor que 0.05, el modelo es globalmente significativo.

### 2.3 Realiza un test de hipotesis para evaluar si las dos variables que se agregaron son significativas para el modelo simultaneamente. Use significancia del 5%.


In [19]:
F_critico <- qf(0.95 , 2 , 203)
R2_r <- 0.03191
R2_ur <- 0.04059
F_est <- (R2_ur - R2_r)/(1-R2_ur)*203/2
if(F_est > F_critico) {"Se rechaza H0"} else {"No se rechaza H0"}


[1] "No se rechaza H0"

### **Interpretación**
No se rechaza la Hipótesis Nula (H0), esto significa que el valor del estadístico F calculado fue menor o igual al valor crítico F. A un nivel de significancia del 5%:

Esto implica que no tenemos suficiente evidencia estadística para concluir que las variables sales_mayor_promedio y sales_mayor_promedio_roe son conjuntamente significativas para explicar el salario del CEO.

### 2.4 Volviendo a la regresión del salario en función de sales, roe y ros. Considere dos grupos, Grupo 1: Aquellos clientes que sales es mayor al promedio, y Grupo 2: Aquellos clientes que sales es menor al promedio. Realice un test de hipotesis para grupos al 5% de significancia. Use el metodo que más le acomode.


In [21]:
ceosal1$sales_D <- ceosal1$sales*ceosal1$sales_mayor_promedio
ceosal1$roe_D <- ceosal1$roe*ceosal1$sales_mayor_promedio
ceosal1$ros_D <- ceosal1$ros*ceosal1$sales_mayor_promedio

modelo <- lm(salary ~ sales+roe+ros+sales_mayor_promedio+sales_D+roe_D+ros_D, data=ceosal1)
summary(modelo)

F_critico <- qf(0.95, 4, 201)
R2_ur <- 0.06028
R2_r <- 0.03191
F_est <- (R2_ur-R2_r)/(1-R2_ur)*201/4
if (F_est > F_critico) {"Se rechaza H0"} else {"No se rechaza H0"}


Call:
lm(formula = salary ~ sales + roe + ros + sales_mayor_promedio + 
    sales_D + roe_D + ros_D, data = ceosal1)

Residuals:
    Min      1Q  Median      3Q     Max 
-1024.7  -456.7  -210.2    83.2 13712.4 

Coefficients:
                       Estimate Std. Error t value Pr(>|t|)  
(Intercept)           331.86179  352.76387   0.941   0.3480  
sales                   0.13362    0.06635   2.014   0.0454 *
roe                    25.63426   13.07068   1.961   0.0512 .
ros                    -0.47570    1.66943  -0.285   0.7760  
sales_mayor_promedio 1043.02106  578.05672   1.804   0.0727 .
sales_D                -0.13033    0.06731  -1.936   0.0542 .
roe_D                 -17.86857   28.02885  -0.638   0.5245  
ros_D                   1.55160    3.84565   0.403   0.6870  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1353 on 201 degrees of freedom
Multiple R-squared:  0.06028,	Adjusted R-squared:  0.02756 
F-statistic: 1.842 on 7 and 201

[1] "No se rechaza H0"

### **Interpretación**
No se rechaza la Hipótesis Nula (H0) en el test de grupos, esto significa que, a un nivel de significancia del 5%, no encontramos evidencia estadística suficiente para concluir que existen diferencias significativas entre los dos grupos (empresas con ventas mayores al promedio y empresas con ventas no mayores al promedio) en cómo los regresores (sales, roe, ros) afectan el salario del CEO.

### 2.5  Usando la regresión del salario en función de sales, roe y ros, se distinguen los grupos mediante las dummies; indus, finance, consprod y utility, que corresponden al tipo de empresa. Realice mediante test de Chow, con 5% de significancia, si los modelos para cada grupo son iguales.

In [22]:
ceosal1_indus <- subset(ceosal1, indus==1)
ceosal1_finance <- subset(ceosal1, finance ==1)
ceosal1_consprod <- subset(ceosal1, consprod==1)
ceosal1_utility <- subset(ceosal1, utility==1)

modelo <- lm(salary ~ sales+roe+ros, data=ceosal1)
modelo_indus <- lm(salary ~ sales+roe+ros, data=ceosal1_indus)
modelo_finance <- lm(salary ~ sales+roe+ros, data=ceosal1_finance)
modelo_consprod <- lm(salary ~ sales+roe+ros, data=ceosal1_consprod)
modelo_utility <- lm(salary ~ sales+roe+ros, data=ceosal1_utility)

SSR_r <- sum(residuals(modelo)^2)
SSR_indus <- sum(residuals(modelo_indus)^2)
SSR_finance <- sum(residuals(modelo_finance)^2)
SSR_consprod <- sum(residuals(modelo_consprod)^2)
SSR_utility <- sum(residuals(modelo_utility)^2)
SSR_ur <- SSR_indus+SSR_finance+SSR_consprod+SSR_utility

G <- 4
k <- 3
n <- nrow(ceosal1)
gl1 <- (G-1)*(k+1)
gl2 <- n-G*(k+1)

F_critico <- qf(0.95, gl1, gl2)
F_est <- (SSR_r-SSR_ur)/SSR_ur*gl2/gl1
if(F_est > F_critico) {"Se rechaza H0"} else {"No se rechaza H0"}

[1] "No se rechaza H0"

### **Interpretación**
No se rechaza la Hipótesis Nula (H0) en el test de Chow. Esto significa que, con un nivel de significancia del 5%, no hay evidencia estadística suficiente para concluir que los modelos de regresión del salario (en función de sales, roe y ros) son diferentes entre los grupos definidos por las variables dummy indus, finance, consprod y utility. En otras palabras, podemos asumir que la relación entre el salario del CEO y las variables sales, roe y ros es la misma para todas las categorías de empresas (industria, finanzas, productos de consumo y servicios públicos).